src/niftyone/io.py

```diff
@@ -24,44 +25,39 @@ def __init__(self, where: StrPath, fps: int):
         self._container = None
         self._stream = None
 
-    def put(self, img: Union[np.ndarray, Image.Image]):
-        """
-        Add a frame to the stream. If this is the first frame, the stream will be
-        initialized.
-        """
+    def put(self, img: Union[np.ndarray, Image.Image]) -> None:
+        """Add frame to the stream."""
         if isinstance(img, np.ndarray):
             img = topil(img)
 
         if self._container is None:
             self.init_stream(width=img.width, height=img.height)
 
         frame = av.VideoFrame.from_image(img)
-        for packet in self._stream.encode(frame):
-            self._container.mux(packet)
+        for packet in self._stream.encode(frame):  # type: ignore [attr-defined]
```

Comment:
Any way to clean up all these type ignores?

Reply:
Turns out you can also disable this through the `pyproject.toml` file. I've done this for `attr-defined` since this was repeated multiple times (I think just for around this block of code). There are 2 other inline `type: ignores` for other types that I've left in.

Reply:
Why not annotate `self._stream` + `self._container` in the constructor? Type can't be inferred because they are `None` on creation.

Reply:
Yeah, that works too - I'll take a closer look at this tomorrow!

Reply:
Instead of ignoring the `mypy` errors here and in 2 other places, addressed them with a minor refactor (can see the details in the latest 3 commits) + explicitly type hinted where needed.